python/servo/platform/linux.py

```diff
@@ -150,17 +150,6 @@ def _platform_bootstrap(self, force: bool) -> bool:
         installed_something |= self._platform_bootstrap_gstreamer(force)
         return installed_something
 
-    def linker_flag(self) -> str:
-        # the rust-lld binary downloaded by rustup
-        # doesn't respect NIX_LDFLAGS and also needs
-        # other patches to work correctly. Use system
-        # version of lld for now. See
-        # https://github.com/NixOS/nixpkgs/issues/220717
-        if self.distro.lower() == 'nixos':
-            return '-C link-arg=-fuse-ld=lld'
```

Comment:
Can we just have this instead (if we detect that lld is installed)?

Reply:
In the end, we'd like to make it so that `./mach build` and `cargo build` are compatible. I think the easiest way toward this is removing all of these non-Cargo-compatible ways of setting rust compiler flags. Rust is also gradually moving toward using its own version of ld to link which should speed up linking when it's enabled. (see https://github.com/rust-lang/rust/issues/39915)

Reply:
Oh, I think it's also interesting to note this doesn't seem to affect compile times on the CI.